### PR TITLE
glimpse for `COLUMN` and `DATAFRAME` completion popups

### PR DESCRIPTION
**Pull request by romainfrancois**
### Intent

addresses #12308

### Approach

`COLUMN`: 

image

image

`DATAFRAME`: (wip display)

image

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by romainfrancois** [comment]
I think `pillar::glimpse()` is robust enough, cc @krlmlr

image

**Comment by krlmlr** [comment]
We can have nested tibbles, packed tibbles, nested lists of various depths. I never benchmarked `glimpse()` for interactive use, happy to help if anything needs to be faster.

**Comment by romainfrancois** [comment]
Still in progress, but we have something like this so far: 

image

image

**Comment by ntluong95** [comment]
Very useful feature

**Comment by romainfrancois** [comment]
Showing more completion items so that we can use more height in the secundary popup: 

image

Is that overwhelming that this shows 9 items, i.e. do we prefer as it was, but the secundary popup takes more space: 

image

**Comment by romainfrancois** [comment]
updated column icon (for when the completion is one column from a dataset) to make it little more different to data frame icon: 

image

**Comment by romainfrancois** [comment]
Trying a different way to show glimpse for columns, having each value from the 6 (?) first results of `pillar::format_glimpse()` in a box. 

image

image

https://user-images.githubusercontent.com/2625526/203106691-64dc7535-63e8-4d1d-9c65-20f11d029812.mov

File: src/cpp/session/modules/SessionHelp.R [hunk 1]
```diff
@@ -438,6 +444,60 @@ options(help_type = "html")
    print(utils::vignette(topic, package))
 })
 
+.rs.addFunction("getHelpColumn", function(name, src, envir = parent.frame())
+{
+   data <- .rs.getAnywhere(src, envir)
+   if (!is.null(data) && isNamespaceLoaded("pillar"))
+   {
+      data <- data[, name, drop = FALSE]
+      text <- capture.output(pillar::glimpse(data, width = 50))
```
[the remark this answers is not in the document]
Comment (romainfrancois) [reply src/cpp/session/modules/SessionHelp.R]: Thanks. This uses `format_glimpse()` now.

**Comment by romainfrancois** [comment]
TODO: `format_glimpse()` does not truncate: 

image

**Comment by romainfrancois** [comment]
Further truncating: 

https://user-images.githubusercontent.com/2625526/203823321-1082d32b-778a-48ee-9ed7-f1682a314147.mov

**Comment by romainfrancois** [comment]
Merging this now, but can/will follow up if needed.

**Pull request merged**